UpdateVisibleServicesActivity/index.ts

```diff
@@ -0,0 +1,211 @@
+﻿import { AzureFunction, Context } from "@azure/functions";
+import { createBlobService } from "azure-storage";
+
+import * as t from "io-ts";
+
+import { Either, isLeft, left, right } from "fp-ts/lib/Either";
+import { isSome, none, Option, some } from "fp-ts/lib/Option";
+
+import { readableReport } from "italia-ts-commons/lib/reporters";
+
+import { ServiceId } from "io-functions-commons/dist/generated/definitions/ServiceId";
+import {
+  VISIBLE_SERVICE_BLOB_ID,
+  VISIBLE_SERVICE_CONTAINER,
+  VisibleService
+} from "io-functions-commons/dist/src/models/visible_service";
+import {
+  acquireLease,
+  getBlobAsObject,
+  releaseLease,
+  upsertBlobFromObject
+} from "io-functions-commons/dist/src/utils/azure_storage";
+import { getRequiredStringEnv } from "io-functions-commons/dist/src/utils/env";
+import { Second } from "italia-ts-commons/lib/units";
+
+// The lease duration in seconds.
+// After the retrive/update activities the lease is released actively by the function.
+// If the function crashes the lease is released by the system.
+const LEASE_DURATION = 15 as Second;
+
+const AddVisibleServiceInput = t.interface({
+  action: t.literal("UPSERT"),
+  visibleService: VisibleService
+});
+
+const RemoveVisibleServiceInput = t.interface({
+  action: t.literal("DELETE"),
+  visibleService: VisibleService
+});
+
+const Input = t.union([AddVisibleServiceInput, RemoveVisibleServiceInput]);
```

Comment:
I was using `union` because `taggedUnion` is now [deprecated](https://github.com/gcanti/io-ts/commit/9f4daf09de016888faab44c7daad6f1c3ce1d224) (union is now able to detect and optimize tagged unions). The problem is that this new feature is available only from v1.9.0 (we are using v1.8.6 that is just before v1.9.0) so i am going to revert the code to use taggedUnion as you suggested.